src/zui/ZUIEditor/EditorOverlays/BlockToolbar/toolbars/BlockToolbarBase.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+  Box,
+  IconButton,
+  Menu,
+  MenuItem,
+  Paper,
+  Typography,
+} from '@mui/material';
+import { FC, useState } from 'react';
+import { FromToProps } from 'remirror';
+import { MoreVert } from '@mui/icons-material';
+import { useCommands } from '@remirror/react';
+
+import MoveUpButton from '../buttons/MoveUpButton';
+import MoveDownButton from '../buttons/MoveDownButton';
+import { Msg } from 'core/i18n';
+import messageIds from 'zui/l10n/messageIds';
+
+type BlockToolbarBaseProps = {
+  conversions?: { label: string; onClick: () => void }[];
+  icon: JSX.Element;
+  range: FromToProps;
+  title: string;
+  tools?: JSX.Element;
```

Comment:
Could this be `children` instead, so that the concrete toolbars don't have to pass JSX to a prop, but instead can just wrap their tools in `<BlockToolbarBase>`?